EventEmitter.d.ts

```diff
@@ -19,30 +19,30 @@ declare namespace EventEmitter {
         [event: string]: Listener[];
     }
 
-    type Listeners = Listener[]|ListenerMap;
+    type Listeners = Listener[] | ListenerMap;
 
-    export class EventEmitter {
+    export class EventEmitter<E = EventKey> {
         static noConflict(): typeof EventEmitter;
 
-        getListeners(event: EventKey): Listeners;
+        getListeners(event: E): Listeners;
         flattenListeners(listeners: Listener[]): void;
-        getListenersAsObject(event: EventKey): ListenerMap;
-        addListener(event: EventKey, listener: Listener|Function): this;
-        on(event: EventKey, listener: Listener|Function): this;
-        addOnceListener(event: EventKey, listener: Function): this;
-        once(event: EventKey, listener: Function): this;
-        defineEvent(event: EventKey): this;
-        defineEvents(events: EventKey[]): this;
-        removeListener(event: EventKey, listener: Function): this;
-        off(event: EventKey, listener: Function): this;
+        getListenersAsObject(event: E): ListenerMap;
+        addListener(event: E, listener: Listener | Function): this;
+        on(event: E, listener: Listener | Function): this;
+        addOnceListener(event: E, listener: Function): this;
+        once(event: E, listener: Function): this;
+        defineEvent(event: E): this;
+        defineEvents(events: E[]): this;
+        removeListener(event: E, listener: Function): this;
+        off(event: E, listener: Function): this;
         addListeners(event: Events, listeners: Function[]): this;
         removeListeners(event: Events, listeners: Function[]): this;
         manipulateListeners(remove: boolean, event: Events, listeners: Function[]): this;
-        removeEvent(event?: EventKey): this;
-        removeAllListeners(event?: EventKey): this;
-        emitEvent(event: EventKey, args?: any[]): this;
-        trigger(event: EventKey, args?: any[]): this;
-        emit(event: EventKey): this;
```

Comment:
To be more clear about the type checking, you must know that everything about the REAL type checking is linke you the the contributors implemeted it. typescript is a layer above javascript that doesnt change javascript code or its behaviour. so to change the *.d.ts file everything in the *.js file will behave like it is originally written. just the tsc (typescript compiler or transpiler) will notice about the changes.